CMakeLists.txt

```diff
@@ -19,6 +19,7 @@ set(CMAKE_CXX_EXTENSIONS ON)
 
 option(ABIEOS_NO_INT128 "disable use of __int128" OFF)
 option(ABIEOS_ONLY_LIBRARY "define and build the ABIEOS library" OFF)
+option(BUILD_SHARED_LIBS "build ABIEOS share library" ON)
```

Comment:
What's the need for making this an option?

Reply:
actually, `BUILD_SHARED_LIBS` is a built-in cmake variable, so odd to redefine an option with same name

Reply:
I cannot get the shared library to build with iPhone SDK and I don't see it actually needed. I just need to way to disable it.

Reply:
From cmake documentation, 

> Global flag to cause add_library() to create shared libraries if on.
> 
> If present and true, this will cause all libraries to be built shared unless the library was explicitly added as a static library. This variable is **often added to projects as an option()** so that each user of a project can decide if they want to build the project using shared or static libraries.

I choose to use BUILD_SHARED_LIBS simply because the `abieos_module` target is just a share library variant of target `abieos` which only export the C APIs. I cannot see anything odd with it. However, if you insist this option should be renamed, I am OK with it. Just tell me the name of option you like. 

Personally, I would prefer to use BUILD_SHARED_LIBS for choosing to build either static or shared library instead of choosing whether to build shared library. Unfortunately, that could affect the dependent projects which is not under my control.

Reply:
This is the same reasoning as the first answer for https://stackoverflow.com/questions/53499678/cmake-build-shared-libs-for-single-libraries ?

Reply:
Use `BUILD_SHARED_LIBS` in this context does have a problem. It's not when build this project by itself, but when it is included as a submodule. `BUILD_SHARED_LIBS` is a global variable rather than a scoped variable. i.e. it can change all the library targets in EOS project. I will change the name of the variable.

Reply:
Still not sure I'm following the reasoning here. `add_library()` without `SHARED` or `STATIC` doesn't imply shared -- it's going to be whatever the global `BUILD_SHARED_LIBS` is set to. So it seems like this option would just be more accurately named `ABIEOS_BUILD_LIB`.

But even that seems unneeded since I'd expect most users to pull in abieos via `EXCLUDE_FROM_ALL` so the library won't be built unless a target depends on it anyways.

Reply:
abieso build both STATIC (target name `abieos`) and SHARED (target name `abieos_module`) libraries. I just tried to provide a way to disable building `abieos_module`.  Using  `ABIEOS_BUILD_LIB` sounds like not any library at all. `EXCLUDE_FROM_ALL` is useful when included as a subproject, not for building as a top project. As I said before, this came up necessary for me to test the cross compiling with iPhone SDK.